### ISSUE TITLE
feat(image_gen): add `denoising_end` (early-termination) for image diffusion

### ISSUE THREAD

**Issue by dahai80** [dahai80/fusion-mlx#846]
## Problem

The image generation denoise loop has no early-termination mechanism:

```python
# fusion_mlx/engines/image_gen.py:874
def _denoise():
    ...
    # image_gen.py:890
    for t in config.time_steps:
        noise = predict(...)
        latents = config.scheduler.step(...)
        mx.eval(latents)
    return latents...
```

There is no `denoising_end` / step-truncation parameter anywhere in the codebase (grep for `denoising_end` returns no matches). The full `num_inference_steps` always runs.

## Why it matters

For use cases like character reference sheets (four-view turnaround panels), the last 7–10% of denoise steps only refine fine noise that is irrelevant for a modeling reference (not a final frame). Truncating those steps saves time with no perceptible quality loss for that purpose. This is a standard knob in diffusers (`denoising_end`) and costs little to add: stop the `for t in config.time_steps` loop once the sigma ratio crosses the threshold.

## Proposal

Add a `denoising_end: float | None = None` param to `generate(...)` (`image_gen.py:510`) and the `/v1/images/generate` request model. When set (0–1), break the denoise loop after the corresponding fraction of the sigma schedule. Default `None` = run all steps (current behavior).

## Environment

- fusion-mlx v0.9.1
- Confirmed by code read 2026-09-07.

## Context

Part of a four-view generation speed-up effort. See companion issue on honoring the `scheduler` param for Qwen-Image.

**Comment by dahai80** [comment]
Implemented in #856 — denoising_end (0-1) truncates sigma schedule. Closing as resolved.
